From 6642a56433236003e53313cdd5c0bded5cf9c2a2 Mon Sep 17 00:00:00 2001
From: Suhas N Bhargav <sbhargav@qti.qualcomm.com>
Date: Fri, 9 Jan 2026 12:33:28 +0530
Subject: [PATCH] net: Fix the ECM notification issue in case of
 (added_by_star_ex) MDB Snooping

This change ensures ECM receives the atomic notification with latest MDB entries when (MDB_PG_FLAGS_STAR_EXCL) flag is set
Signed-off-by: Suhas N Bhargav <sbhargav@qti.qualcomm.com>

Change-Id: I95ffd974a5f5cbc3e23858ac9d25bc8e852957a6
---
 net/bridge/br_mdb.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/net/bridge/br_mdb.c b/net/bridge/br_mdb.c
index 412a20aa82bf..c913db91e6f2 100644
--- a/net/bridge/br_mdb.c
+++ b/net/bridge/br_mdb.c
@@ -607,10 +607,10 @@ void br_mdb_notify(struct net_device *dev,
 
 #if IS_ENABLED(CONFIG_BRIDGE_MCAST_OFFLOAD)
 	/*
-	 * Notification should only be generated for a (*,G) MDB entry
-	 * having a valid Port-Group pointer.
+	 * Notification should be generated for a (*,G) MDB entry or
+	 * an entry with STAR_EXCL flag, having a valid Port-Group pointer.
 	 */
-	if (br_multicast_is_star_g(&mp->addr) && pg)
+	if (pg && (br_multicast_is_star_g(&mp->addr) || (pg->flags & MDB_PG_FLAGS_STAR_EXCL)))
 		ret = br_mcast_offload_mdb_send_event_notify(dev, mp, type);
 
 	if (ret)
-- 
2.34.1

